eth_dummy_phy/src/esp_eth_phy_dummy.c

```diff
@@ -0,0 +1,188 @@
+/*
+ * SPDX-FileCopyrightText: 2023-2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <string.h>
+#include <stdlib.h>
+#include <sys/cdefs.h>
+#include "esp_log.h"
+#include "esp_check.h"
+#include "esp_eth_driver.h"
+#include "driver/gpio.h"
+
+typedef struct {
+    esp_eth_phy_t parent;
+    esp_eth_mediator_t *eth;
+    int reset_gpio_num;
+    eth_link_t link;
+    eth_speed_t speed;
+    eth_duplex_t duplex;
+} phy_dummy_t;
+
+static const char *TAG = "dummy_phy";
+
+static esp_err_t get_link(esp_eth_phy_t *phy)
+{
+    esp_err_t ret = ESP_OK;
+    phy_dummy_t *dummy_phy = __containerof(phy, phy_dummy_t, parent);
+    esp_eth_mediator_t *eth = dummy_phy->eth;
+
+    // TODO: check if EMAC is working (e.g. External CLK is present)
+
+    if (dummy_phy->link == ETH_LINK_DOWN) {
+        dummy_phy->link = ETH_LINK_UP;
+
+        ESP_GOTO_ON_ERROR(eth->on_state_changed(eth, ETH_STATE_SPEED, (void *)dummy_phy->speed), err, TAG, "change speed failed");
+        ESP_GOTO_ON_ERROR(eth->on_state_changed(eth, ETH_STATE_DUPLEX, (void *)dummy_phy->duplex), err, TAG, "change duplex failed");
+
+        bool peer_pause_ability = false; // always false
+        ESP_GOTO_ON_ERROR(eth->on_state_changed(eth, ETH_STATE_PAUSE, (void *)peer_pause_ability), err, TAG, "change pause ability failed");
+
+        ESP_GOTO_ON_ERROR(eth->on_state_changed(eth, ETH_STATE_LINK, (void *)dummy_phy->link), err, TAG, "change link failed");
+    }
+
+err:
+    return ret;
+}
+
+static esp_err_t set_link(esp_eth_phy_t *phy, eth_link_t link)
+{
+    esp_err_t ret = ESP_OK;
+    phy_dummy_t *dummy_phy = __containerof(phy, phy_dummy_t, parent);
+    esp_eth_mediator_t *eth   = dummy_phy->eth;
+
+    if (dummy_phy->link != link) {
+        dummy_phy->link = link;
+        // link status changed, inmiedately report to upper layers
+        ESP_GOTO_ON_ERROR(eth->on_state_changed(eth, ETH_STATE_LINK, (void *)dummy_phy->link), err, TAG, "change link failed");
+    }
+err:
+    return ret;
+}
+
+static esp_err_t set_mediator(esp_eth_phy_t *phy, esp_eth_mediator_t *eth)
+{
+    esp_err_t ret = ESP_OK;
+    ESP_GOTO_ON_FALSE(eth, ESP_ERR_INVALID_ARG, err, TAG, "mediator can't be null");
+    phy_dummy_t *dummy_phy = __containerof(phy, phy_dummy_t, parent);
+    dummy_phy->eth = eth;
+err:
+    return ret;
+}
+
+static esp_err_t reset_hw(esp_eth_phy_t *phy)
+{
+    phy_dummy_t *dummy_phy = __containerof(phy, phy_dummy_t, parent);
+
+    if (dummy_phy->reset_gpio_num >= 0) {
```

Comment:
Do we want to preform a software reset? Not sure what's the function of the reset gpio used by a dummy phy :thinking:

Reply:
User could configure it to reset the other device in similar way as PHY if they find it useful.